src/ValueParsers/DateFormatParser.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+namespace ValueParsers;
+
+use DataValues\TimeValue;
+
+/**
+ * This parser is in essence the inverse operation of Language::sprintfDate.
+ *
+ * @see Language::sprintfDate
+ *
+ * @since 0.8.1
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+class DateFormatParser extends StringValueParser {
+
+	const FORMAT_NAME = 'datetime';
+
+	const OPT_DATE_FORMAT = 'dateFormat';
+	const OPT_DIGIT_TRANSFORM_TABLE = 'digitTransformTable';
```

Comment:
"Array of strings, mapping canonical digit characters ("1", "2" and so on, possibly including "." and ",") to localized characters."